packages/openapi/src/openapi-request-builder.ts

```diff
@@ -164,23 +164,36 @@ export class OpenApiRequestBuilder<ResponseT = any> {
 
   private getPath(): string {
     const pathParameters = this.parameters?.pathParameters || {};
+    let path = this.pathPattern;
+    for (const paramName in pathParameters) {
+      if (!path.includes('{' + paramName + '}')) {
+        throw new Error(
+          `Cannot execute request, no path parameter '${paramName}' found in the path pattern.`
+        );
+      }
+      const paramValue = String(pathParameters[paramName]);
+      if (/[/?#]+/.test(paramValue)) {
+        throw new Error(
+          `Cannot execute request, the value of the path parameter '${paramName}' must not contain '/', '?', or '#'. (RFC3986)`
+        );
+      }
+      path = path.replace(
+        new RegExp(`{${paramName}}`, 'g'),
+        encodeURIComponent(paramValue)
+      );
+    }
 
-    const pathParts = this.pathPattern.split('/');
-    return pathParts
-      .map(part => {
-        if (OpenApiRequestBuilder.isPlaceholder(part)) {
-          const paramName = part.slice(1, -1);
-          const paramValue = pathParameters[paramName];
-          if (!paramValue) {
-            throw new Error(
-              `Cannot execute request, no path parameter provided for '${paramName}'.`
-            );
-          }
-          return encodeURIComponent(paramValue);
-        }
-        return part;
-      })
-      .join('/');
+    // Check if there is still curly bracket in the replaced path pattern
```

Comment:
Please let me know if check at this step is needed. Currently, if there is any `{...}` left, an error will be thrown.